File: C++/2962. Count Subarrays Where Max Element Appears at Least K Times.cpp
class Solution {
public:
    long long int countSubs(vector<int>& nums, int k, int mx) {
        long long int count = 0;
        int left = 0;
        unordered_map<int, int> freq;

        for (int right = 0; right < nums.size(); ++right) {
            
            freq[nums[right]]++;
            
            while (freq[mx] >= k) {
                freq[nums[left]]--;
                left++;                   
            }
            
            count += right - left + 1;
                    
        }
        return count;
    }
    long long countSubarrays(vector<int>& nums, int k) {
        int mx = -1;
        for(int m: nums)
        {
            mx = max(mx, m);
        }
        long long int Tsubs = nums.size()*(nums.size()+1)/2;
    
        return (long long) Tsubs - countSubs(nums, k, mx);
       
       
    }
};

File: JAVA/2962. Count Subarrays Where Max Element Appears at Least K Times.java
class Solution {
    public long countSubarrays(int[] nums, int k) {
        int max=0;
        int count=0;
        long len=0;
        int i=0;
        int j=0;
        for(int n:nums)max=Math.max(n,max);
        while(j<nums.length){
            int n=nums[j];
            if(n==max)count++;
            while(count>=k){
                if(nums[i]==max)count--;
                i++;
            }
            len+=j-i+1;
            j++;
        }
       long total = ((long) nums.length * (nums.length + 1)) / 2;
        return  total-len;
    }
}
